src/library_webgpu.js

```diff
@@ -958,11 +996,74 @@ var LibraryWebGPU = {
     var pass = WebGPU.mgrRenderPassEncoder.get(passId);
     pass.drawIndexed(indexCount, instanceCount, firstIndex, baseVertex, firstInstance);
   },
+  dawnRenderPassEncoderExecuteBundles: function(passId, count, bundlesPtr) {
+    var pass = WebGPU.mgrRenderPassEncoder.get(passId);
+
+#if ASSERTIONS
+    assert(bundlesPtr % 4 === 0);
+#endif
+
+    var bundles = Array.from(HEAP32.subarray(bundlesPtr >> 2, (bundlesPtr >> 2) + count),
+      function(id) { return WebGPU.mgrRenderBundle.get(id); });
+    pass.executeBundles(bundles);
+  },
   dawnRenderPassEncoderEndPass: function(passId) {
     var pass = WebGPU.mgrRenderPassEncoder.get(passId);
     pass.endPass();
   },
 
+  // Render bundle encoder
+
+  dawnRenderBundleEncoderSetBindGroup: function(bundleId, groupIndex, groupId, dynamicOffsetCount, dynamicOffsetsPtr) {
+    var pass = WebGPU.mgrRenderBundleEncoder.get(bundleId);
+    var group = WebGPU.mgrBindGroup.get(groupId);
+    if (dynamicOffsetCount == 0) {
+      pass.setBindGroup(groupIndex, group);
+    } else {
+      var offsets = [];
+      for (var i = 0; i < dynamicOffsetCount; i++, dynamicOffsetsPtr += 8) {
+        offsets.push({{{ gpu.makeGetU64('dynamicOffsetsPtr', 0) }}});
```

Comment:
This will silently break when dynamicOffsetsPtr becomes 32-bit. Leave a TODO, at least.

Reply:
Done.